unit-tests/src/test/java/org/eclipse/collections/impl/factory/MultimapsTest.java

```diff
@@ -74,6 +76,21 @@ public void immutableSortedSet()
         Assert.assertEquals(TreeSortedSetMultimap.newMultimap(Tuples.pair(1, 1), Tuples.pair(2, 2), Tuples.pair(3, 3)), three);
     }
 
+    @Test
+    public void immutableSortedBagMultimap()
+    {
+    	ImmutableSortedBagMultimap<Integer, Integer> empty = Multimaps.immutable.multimap.of(Integer::compareTo);
+    	ImmutableSortedBagMultimap<Integer, Integer> emptyWith = Multimaps.immutable.multimap.with(Integer::compareTo);
+        Verify.assertEmpty(empty);
+        Verify.assertEmpty(emptyWith);
+        ImmutableSortedBagMultimap<Integer, Integer> one = Multimaps.immutable.multimap.of(Integer::compareTo,1, 1);
+        Assert.assertEquals(TreeBagMultimap.newMultimap(Tuples.pair(1, 1)), one);
+        ImmutableSortedBagMultimap<Integer, Integer> two = Multimaps.immutable.multimap.of(Integer::compareTo,1, 1, 2, 2);
+        Assert.assertEquals(TreeBagMultimap.newMultimap(Tuples.pair(1, 1), Tuples.pair(2, 2)), two);
+        ImmutableSortedBagMultimap<Integer, Integer> three = Multimaps.immutable.multimap.of(Integer::compareTo,1, 1, 2, 2, 3, 3);
+        Assert.assertEquals(TreeBagMultimap.newMultimap(Tuples.pair(1, 1), Tuples.pair(2, 2), Tuples.pair(3, 3)), three);
```

Comment:
It would be good to test using type variance too. Sort by Object::toString for example, but store subclasses like Integer.

Reply:
what is your opnion about that , fit in your suggestion?

 ImmutableSortedBagMultimap<? extends Object,? extends Object> toString = Multimaps.immutable.multimap.of(String::compareTo, "A", "B");
        Assert.assertEquals(TreeBagMultimap.newMultimap(Tuples.pair("A", "B")), toString);